src/Sylius/Bundle/ResourceBundle/Controller/FlashHelper.php

```diff
@@ -48,38 +40,66 @@ public function __construct(SessionInterface $session, TranslatorInterface $tran
     public function addSuccessFlash(RequestConfiguration $requestConfiguration, $actionName, ResourceInterface $resource = null)
     {
         $metadata = $requestConfiguration->getMetadata();
-        $flashMessage = $requestConfiguration->getFlashMessage($actionName);
+        $message = $requestConfiguration->getFlashMessage($actionName);
 
-        if (false === $flashMessage) {
+        if (false === $message) {
             return;
         }
 
-        $translatedMessage = $this->translateMessage($flashMessage, $metadata);
-
-        if ($flashMessage === $translatedMessage) {
-            $translatedMessage = $this->translateMessage(sprintf('sylius.resource.%s', $actionName), $metadata);
+        if ($this->isResourceDefaultMessage($message, $metadata->getName(), $actionName)) {
+            $message = $this->getResourceMessage($actionName);
```

Comment:
1. We've got a message `sylius.customer.update`
2. It happens to be the resource default message
3. We replace it with `sylius.resource.update`
4. No profit, wtf?